### ISSUE TITLE
Catch KeyError and Raise BadRequestError for Invalid phone_info

### ISSUE THREAD

**Issue by k-macmillan** [department-of-veterans-affairs/notification-api#1525]
## User Story - Business Need
We respond with a 500 if an SMS request is made with an invalid prefix. The 500 is raised because we throw a KeyError in the utils repo and do not handle it properly. We should always return a 2xx or 4xx to services.

- [x] Ticket is understood, and QA has been contacted (if the ticket has a QA label).

### User Story(ies)

**As a** service
**I want** my request to not return server errors from VA Notify
**So that** I know why my request failed

### Additional Info and Resources

Logging:
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBL09lQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--041dcf549ddf78e7d32d88be1210ad4cc2e2a993/image.png)

This is the offending code and location (`get_international_phone_info` is the method that raises the `KeyError`):
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNzZmQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--b707144fe6d4c73569bfc2e7f5dcab1bd0b19f8c/image.png)

## Engineering Checklist

- [x] Call wrapped in a try/except block
- [x] KeyError caught and BadRequestError with appropriate message is raised
- [x] Unit test added

## Acceptance Criteria

- [x] Calling the sms endpoint with an invalid prefix does not result in a 500 status code
- [x] The response allows the requestor to understand what the failure was

## QA Considerations
- [ ] Test added to regression suite (maybe?)

## Out of Scope
Any modification to notification-utils

**Comment by mjones-oddball** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/VA-Notify-620d21369d810a00146ed9c8/issues/department-of-veterans-affairs/notification-api/1525?planning-poker) with Zenhub @cris-oddball @EvanParish @k-macmillan @kalbfled @ldraney @nikolai-efimov

**Comment by k-macmillan** [comment]
Pulled this in to start. Update tickets are going to be used for training.

**Comment by k-macmillan** [comment]
Was having difficulty replicating a failing example. Eventually found a number combination that would result in raising the error we have been seeing. PR is up. Tests pass, regression passes.

**Comment by cris-oddball** [comment]
PR approved and merged, tag cut for it and sent to Perf. Will test using the same number in the unit test and add a test to the regression, so this may sit here a day while I get caught up.

**Comment by cris-oddball** [comment]
**PASSED** v2 SMS

Using the same phone number in the unit tests in POST sms, received the expected response from the GET notifications route
```
{
    "template_id": "{{sms-template-id}}",
    "phone_number": "+80888888888"
}
```

400 response
```
{
    "errors": [
        {
            "error": "BadRequestError",
            "message": "Invalid International Billing Prefix"
        }
    ],
    "status_code": 400
}
```

**Comment by cris-oddball** [comment]
Holding this ticket until I can write a test in the regression suite.

**Comment by cris-oddball** [comment]
PR written for this code change, however I cannot merge that PR until we have master up on staging (or else the test will fail). Closing this ticket and will make a note to handle opening the PR for review after the next staging deploy.
